src/librustc_parse/parser/stmt.rs

```diff
@@ -471,11 +450,7 @@ impl<'a> Parser<'a> {
         Ok(Some(stmt))
     }
 
-    fn warn_missing_semicolon(&self) {
-        self.diagnostic().struct_span_warn(self.token.span, {
-            &format!("expected `;`, found {}", self.this_token_descr())
-        }).note({
-            "this was erroneously allowed and will become a hard error in a future release"
-        }).emit();
```

Comment:
Where did this go? We can't suddenly start rejecting code, even if we were unconditionally warning on it.

Reply:
It was always intended to become an error eventually and has been a warning for 3 years, so I figured we can turn it into a hard error by now.

Reply:
SGTM, make sure to run crater :)

Reply:
SGTM, make sure to run crater :)

Reply:
I'll remove it from the PR for now so we don't have to wait for crater and so I don't have to rebase.
I've filed https://github.com/rust-lang/rust/issues/67098 as a reminder.

Reply:
Done.